examples/components/utils/aggregatemodelweights/run.py

```diff
@@ -165,7 +176,24 @@ def main(cli_args=None):
     for model_path in model_paths:
         model_handler.add_model(model_path)
 
-    model_handler.save_model(os.path.join(args.output, f"model.{args.extension}"))
+    # check if meta data needed, coppy the whole dir
+    if args.ancillary_files:
+        for src in os.listdir(args.checkpoints[0]):
```

Comment:
why are we only saving metadata of  `silo_0`?

Reply:
I think the tokenizer files and even hyperparams could differ for each silo. Wouldn't that be the case? As silo local trainings are independent of each other. (Note: It might not be the case in your current scenario but if we want to generalize it then this is also possible, right?)

Reply:
For the tokenizer files, here is an example:
{
  "bos_token": "[CLS]",
  "cls_token": "[CLS]",
  "eos_token": "[SEP]",
  "mask_token": "[MASK]",
  "pad_token": "[PAD]",
  "sep_token": "[SEP]",
  "unk_token": "[UNK]"
}
I think as long as these specifications are consistent for the datasets used in different silos, they should be the same. For the hyper parameter files, I agree they could be different for each silo. I will change the component to accommodate that! Thanks!

Reply:
I think your point Amit is very valid, we might have different metadata in different silos. Although it might be a minority of use cases. Would it be ok to merge the PR with its current design (once other comments are resolved) and investigate a more flexible design in another PR?

Reply:
Sounds good to me! We can address this in another PR